js/src/launchAsync.ts

```diff
@@ -63,6 +74,259 @@ let launchResponseResolve: LaunchResponseResolve;
 let launchResponseReject: LaunchReject;
 let launchResponseError: Error;
 
+// Added globals to persist between functions
+let timeoutId: number | null = null;
+let parent: Node = null;
+let iframeContainer: HTMLDivElement = null;
+let iframe: HTMLIFrameElement = null;
+let noscroll: HTMLStyleElement = null;
+let messageHandler: (e: any) => void = null;
+
+enum FuncType { launchAsync, launchWithoutContentAsync }
+
+function _initializeOptions(options?: Options): Options {
+
+    const { allowFullscreen, cookiePolicy, hideExitButton, onExit, parent, timeout, uiZIndex, useWebview } = options || {};
+
+    const _options = {
+        ...options,
+        uiZIndex: (!uiZIndex || typeof options.uiZIndex !== 'number') ? 1000 : uiZIndex, // Default to 1000 if not valid
+        timeout: timeout ?? 15000,  // Default to 15 seconds
+        useWebview: useWebview || false,
+        allowFullscreen: allowFullscreen || true,
+        hideExitButton: hideExitButton || false,
+        cookiePolicy: cookiePolicy ?? CookiePolicy.Disable, // Default to disable as presently
+        onExit
+    };
+
+    _generateIframeVariables(parent, useWebview, allowFullscreen);
```

Comment:
kind of weird that is call is in a method called _initializeOptions(), but it has noting to do with that